pkg/v3/plugin/ocr3.go

```diff
@@ -7,19 +7,27 @@ import (
 	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
 	"github.com/smartcontractkit/libocr/offchainreporting2plus/types"
 
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
 	ocr2keepersv3 "github.com/smartcontractkit/ocr2keepers/pkg/v3"
 )
 
 type InstructionStore interface{}
 
 type SamplingStore interface{}
 
+type Encoder interface {
+	Encode(...ocr2keepers.CheckResult) ([]byte, error)
+}
+
 type ocr3Plugin[RI any] struct {
-	PrebuildHooks     []func(ocr2keepersv3.AutomationOutcome) error
-	BuildHooks        []func(*ocr2keepersv3.AutomationObservation, InstructionStore, SamplingStore, ocr2keepersv3.ResultStore) error
-	InstructionSource InstructionStore
-	MetadataSource    SamplingStore
-	ResultSource      ocr2keepersv3.ResultStore
+	PrebuildHooks      []func(ocr2keepersv3.AutomationOutcome) error
+	BuildHooks         []func(*ocr2keepersv3.AutomationObservation, InstructionStore, SamplingStore, ocr2keepersv3.ResultStore) error
```

Comment:
v3.InstructionStore etc

Reply:
This should be part of another PR coming through.